src/zenml/integrations/hyperai/service_connectors/hyperai_service_connector.py

```diff
@@ -0,0 +1,386 @@
+                paramiko.AutoAddPolicy()
+            )
+            paramiko_client.connect(
+                hostname=ip_address,
+                username=self.config.username,
+                pkey=paramiko_key,
+                timeout=30,
+            )
+
+            return paramiko_client
+
+        except paramiko.ssh_exception.BadHostKeyException as e:
+            logger.error("Bad host key: %s", e)
+        except paramiko.ssh_exception.AuthenticationException as e:
+            logger.error("Authentication failed: %s", e)
+        except paramiko.ssh_exception.SSHException as e:
+            logger.error(
+                "SSH error: %s. A common cause for this error is selection of the wrong key type in your service connector.",
+                e,
+            )
+        except Exception as e:
+            logger.error(
+                "Unknown error while connecting to HyperAI instance: %s. Please check your network connection, IP address, and authentication details.",
+                e,
+            )
+
+        return None
+
+    def _authorize_client(self) -> None:
+        """Verify that the client can authenticate with the HyperAI instance.
+
+        Raises:
+            BadHostKeyException: If the host key is invalid.
+            AuthenticationException: If the authentication credentials are
+                invalid.
+            SSHException: If there is an SSH related error.
+        """
+        logger.info("Verifying connection to HyperAI instance...")
+
+        try:
+            paramiko_client = self._create_paramiko_client()
+            if paramiko_client is None:
+                raise RuntimeError(
+                    "Could not authorize client due to missing paramiko client."
+                )
+            else:
+                paramiko_client.close()
+        except Exception as e:
+            logger.error(
+                "Unknown error while connecting to HyperAI instance: %s. Please check your network connection, IP address, and authentication details.",
+                e,
+            )
+
+    def _connect_to_resource(
+        self,
+        **kwargs: Any,
+    ) -> Any:
+        """Connect to a HyperAI instance. Returns an authenticated SSH client.
+
+        Args:
+            kwargs: Additional implementation specific keyword arguments to pass
+                to the session or client constructor.
+
+        Returns:
+            An authenticated Paramiko SSH client.
+        """
+        logger.info("Connecting to HyperAI instance...")
+        assert self.resource_id is not None
+
+        try:
+            paramiko_client = self._create_paramiko_client()
+            if paramiko_client is None:
+                raise RuntimeError(
+                    "Could not connect to instance due to missing paramiko client."
+                )
+            else:
+                return paramiko_client
+        except Exception as e:
+            logger.error(
```

Comment:
same here: this should be raised as an `AuthorizationException`.

Reply:
I have removed the whole `try/except` block in ae7b069d4 because it is no longer necessary: `_create_paramiko_client` will raise the `AuthorizationException` if no client can be created.